custom/icds_reports/filters.py

```diff
@@ -38,12 +38,22 @@ def loc_to_json(loc, project):
     project = Domain.get_by_name(domain)
 
     locations = SQLLocation.root_locations(domain)
+
+    if user.has_permission(domain, 'access_all_locations'):
+        all_user_locations = []
+    else:
+        user_location_list = SQLLocation.objects.accessible_to_user(domain, user)
+        user_location_list = [loc.location_id for loc in user_location_list]
+        user_location_ancestors = [loc.location_id for loc in user.sql_location.get_ancestors(include_self=True)]
+        all_user_locations = user_location_list + user_location_ancestors
```

Comment:
I think this can be further improved. Suggesting something that should be tested/confirmed
```
user_has_access_to_all_locations = user.has_permission(domain, 'access_all_locations')
if user_has_access_to_all_locations:
    accessible_root_locations = SQLLocation.root_locations(domain)
else:
    accessible_root_locations = user.sql_location.get_ancestors_of_type(type_code='state')
    
    # ...

    loc_json = [
        loc_to_json(loc, project) for loc in accessible_root_locations
    ]

# ....
if selected_location_id:
    if not user_has_access_to_all_locations:
        primary_location = user.sql_location
        accessible_ancestor_ids = primary_location.get_ancestors().values_list('location_id', flat=True)
        accessible_descendant_ids = primary_location.get_descendants().values_list('location_id', flat=True)
        accessible_location_ids = set(primary_location.location_id + accessible_ancestor_ids + accessible_descendant_ids)
```

Reply:
incorporated